src/components/evm/inputs/CurrencyInput.vue

```diff
@@ -0,0 +1,1121 @@
+        },
+
+        // if the user has clicked into and then out of the input, the input is considered dirty
+        handleBlur() {
+            this.inputIsDirty = true;
+        },
+
+        // handle when the user clicks the maximum available amount above the input
+        fillMaxValue() {
+            if (this.isDisabled || this.isReadonly || !this.maxValue || this.maxValue.eq(this.modelValue)) {
+                return;
+            }
+
+            let formattedMaxValue: string;
+
+            // determine what text should be shown in the input; if the user has swapped currencies,
+            // show the max value in the secondary currency
+            if (this.currenciesAreSwapped) {
+                const maxValueInSecondaryCurrency = convertCurrency(
+                    this.maxValue,
+                    this.decimals,
+                    this.secondaryCurrencyDecimals,
+                    this.secondaryCurrencyConversionFactor,
+                );
+                const maxValueDecimals = formatUnits(maxValueInSecondaryCurrency, this.secondaryCurrencyDecimals).split('.')[1].length;
+
+                formattedMaxValue = prettyPrintCurrency(
+                    maxValueInSecondaryCurrency,
+                    maxValueDecimals,
+                    this.locale,
+                    false,
+                    undefined,
+                    undefined,
+                    this.secondaryCurrencyDecimals,
+                    true,
+                );
+            } else {
+                const maxValueDecimals = formatUnits(this.maxValue, this.decimals).split('.')[1].length;
+
+                formattedMaxValue = prettyPrintCurrency(
+                    this.maxValue,
+                    maxValueDecimals,
+                    this.locale,
+                    undefined,
+                    undefined,
+                    undefined,
+                    this.decimals,
+                    true,
+                );
+            }
+
+            this.setInputValue(formattedMaxValue);
+            this.handleInput();
+        },
+
+        // handle when the user clicks the secondary currency label under the input
+        handleSwapCurrencies() {
+            if (!this.isReadonly && !this.isDisabled) {
+                this.swapCurrencies = !this.swapCurrencies;
+            }
+        },
+
+        // focus the literal input element; may be useful to call from outside the component
+        focusInput() {
+            if (this.inputElement) {
+                this.inputElement.focus();
+            }
+        },
+
+
+        // can be called from outside the component to show an error state for empty input, e.g. when submitting form
+        // without filling out required fields. Has no effect unless the component is set to required
+        showEmptyError() {
+            this.inputIsDirty = true;
+        },
+    },
+});
+</script>
+
+<template>
+<div
+    :id="$attrs.id"
+    :class="{
+        [$attrs.class]: true,
```

Comment:
Here I see two problems:

- First, the IDE is not happy with the typing
![image](https://github.com/telosnetwork/telos-wallet/assets/4420760/d9a8b61d-424a-49d8-a838-d68f6ae79cc8)

- Second, currently, it applies an "undefined" class to the component when the class attribute is not used on the component. 
![image](https://github.com/telosnetwork/telos-wallet/assets/4420760/26444077-0062-4bbb-b04a-ec88c8af5d2e)

the following change fix the two.

```suggestion
    :id="($attrs.id as string)"
    :class="{
        [$attrs.class as string]: !!$attrs.class,
```